### DISCUSSION TITLE
How can I share files between my macOS host and macOS guest?

### DISCUSSION THREAD

**Discussion by ghost** [utmapp/UTM#5278] [macOS]
I have downloaded big files of software onto my macOS host that I want to be **accessible/shared** to my macOS 12.6 guest.

I have tried enabling the "File Sharing" service in System Preferences > Sharing, and it appears in the "Network" part in Finder, but I am unable to connect to it. The same happens when I enter the guest's IP address in the "Cmd + K" menu in Finder.

i need help :(

**Comment by elton2048** [comment]
## Mount sharing folder using virtiofs (Virtualization framework from macOS)

After checking the macOS dev docs and also some code in UTM project I figured out the way to mount the Shared Directories set in macOS VM

1. Set the Shared Path in Sharing section in option section of the VM
2. Run your VM guest
3. In terminal, run command `mount_virtiofs share [mount_path]`

You should now find the Shared Directories in [mount_path]. Note the command in https://docs.getutm.app/guest-support/macos/ should be currently wrong so it won't mount correctly.

Remark: https://github.com/utmapp/UTM/blob/d5a9811f0c926da5ba67ada131029cf344a25d2d/Configuration/UTMAppleConfiguration.swift#L254 

This is where the `share`(tag) comes from.

---

## File sharing feature in System

By your description you may also want to share files using `smb` through the macOS File Sharing feature. The way to do that is as follow

1. Enable File Sharing feature in host machine
2. Set the authorization policy and check the address from the prompt, it should in the format of`smb://xxx.xxx.xxx.xxx`
3. Connect the address In guest machine, just like what you said, "Cmd + K" in Finder and type the host address to connect.
4. Finish the authorization process by what you set in previous step

Hope this helps people using File Sharing in guest macOS.
